src/Hal/Report/Csv/Reporter.php

```diff
@@ -40,30 +45,27 @@ public function generate(Metrics $metrics)
         if (!$logFile) {
             return;
         }
-        if (!file_exists(dirname($logFile)) || !is_writable(dirname($logFile))) {
-            throw new \RuntimeException('You don\'t have permissions to write CSV report in ' . $logFile);
+
+        $logDir = dirname($logFile);
+        if (!file_exists($logDir) || !is_writable($logDir)) {
+            throw new RuntimeException('You do not have permissions to write CSV report in ' . $logFile);
         }
 
-        $availables = (new Registry())->allForStructures();
-        $hwnd = fopen($logFile, 'w');
-        fputcsv($hwnd, $availables);
+        $availableMetrics = (new Registry())->allForStructures();
+        $logPointer = fopen($logFile, 'wb');
```

Comment:
why `b` ? That's not a [standard flag for fopen](https://www.php.net/manual/fr/function.fopen.php)

Reply:
I always use `b` for portability, as PHP itself recommend: 
```
For portability, it is strongly recommended that you always use the 'b' flag when opening files with fopen().
```

Reply:
I didn't known. Ok :)